### PR TITLE
fix(provider): don't query leases on startup

### PR DESCRIPTION
**Pull request by boz**
`ActiveLeasesForProvider` never terminates; guard
against instantiating closed deployments instead

supersedes #1542

### CODE REVIEW

**Comment by boz** [comment]
There is an issue here where when a manifest is submitted, it is assumed that the group data is present.

**Comment by boz** [comment]
never mind, it "just works" - requests are queued until the data is fetched.

File: provider/cluster/service.go [hunk 1]
```diff
@@ -360,37 +359,11 @@ func (s *service) teardownLease(lid mtypes.LeaseID) {
 	}
 }
 
-func findDeployments(ctx context.Context, log log.Logger, client Client, session session.Session) ([]ctypes.Deployment, error) {
+func findDeployments(ctx context.Context, log log.Logger, client Client, _ session.Session) ([]ctypes.Deployment, error) {
```
Comment (hydrogen18) [inline provider/cluster/service.go:362]: Can't we just remove this method?

Comment (boz) [reply provider/cluster/service.go:362]: yeah

Comment (boz) [reply provider/cluster/service.go:362]: I kept it as a placeholder for something that looks in CRDs/etcd but I can remove it.

Comment (hydrogen18) [reply provider/cluster/service.go:362]: OK, I think the same goes for the other method below

File: provider/cluster/manager.go [hunk 2]
```diff
@@ -7,6 +7,7 @@ import (
 	"time"
 
 	"github.com/boz/go-lifecycle"
+	"github.com/pkg/errors"
```
Comment (hydrogen18) [inline provider/cluster/manager.go:10]: Just import `"errors"`.

**Review by dmikey — APPROVED** [review]

**Review by hydrogen18 — APPROVED** [review]

**Pull request merged**